src/ripple/protocol/Seed.h

```diff
@@ -115,8 +116,13 @@ template <>
 boost::optional<Seed>
 parseBase58 (std::string const& s);
 
+/** Same as above, but inculde optional keyType
+*/
+boost::optional<std::pair<Seed, boost::optional<KeyType>>>
+parseBase58Seed (std::string const& s);
+
 /** Attempt to parse a string as a seed */
-boost::optional<Seed>
+boost::optional<std::pair<Seed, boost::optional<KeyType>>>
```

Comment:
Ugh, I'm not sure I like an `optional` that contains a `pair` that contains an `optional`. It's like Russian nesting dolls. I don't have a better alternative to offer. @HowardHinnant, @scottschurr?

Reply:
I considered making `optional<KeyType>` a member of `Seed`, but decided to keep it out of the class (as most of the time it will be empty). I also didn't want another struct of a `Seed + KeyType` as we already have a `Seed`. If there are objections to this interface I'll add it to the `Seed` class. But I'll also wait for other suggestions before I do so.

Reply:
Addressed in a623be578ec6ef192a09cae3d123dcdb19f374d0

Note: I addressed it by adding an `optional<KeyType>` member to seed. As you know, seeds shouldn't have KeyTypes. A single seed should be able to generate multiple keys of multiple types. However, that isn't how it's used. Given the current conflating of seeds and keys I think the cleanest solution is to add the `optional<KeyType` to seed.